src/coreclr/nativeaot/Runtime/FinalizerHelpers.cpp

```diff
@@ -29,14 +29,25 @@ CLREventStatic g_FinalizerDoneEvent;
 
 static HANDLE g_lowMemoryNotification = NULL;
 
+#ifdef TARGET_WINDOWS
+static bool g_FlsInitSucceeded = false;
+#endif
+
 EXTERN_C void QCALLTYPE ProcessFinalizers();
 
-// Unmanaged front-end to the finalizer thread. We require this because at the point the GC creates the
-// finalizer thread we can't run managed code. Instead this method waits
+// Unmanaged front-end to the finalizer thread. We require this because at the point when this thread is
+// created we can't run managed code. Instead this method waits
 // for the first finalization request (by which time everything must be up and running) and kicks off the
 // managed portion of the thread at that point
 uint32_t WINAPI FinalizerStart(void* pContext)
 {
+#ifdef TARGET_WINDOWS
+    g_FlsInitSucceeded = PalInitComAndFlsSlot();
```

Comment:
Do we need to exit the finalizer thread when this fails? Otherwise, we call `ThreadStore::AttachCurrentThread` below that is going to call `FlsSetValue` on uninitialized `g_flsIndex` and things will just go bad from there.